src/views/GameListView.jsx

```diff
@@ -112,24 +180,59 @@ const GameListView = () => {
       })
   );
 
-  const handleChange = (event) => {
-    setSearchTerm(event.target.value);
-  };
-
   const filteredGameList = React.useMemo(() => {
     return gameList
-      ? gameList.filter(
-          (game) =>
-          searchTerm === NULL_SEARCH_TERM ||
-            cleanWord(game.defaultName).includes(cleanWord(searchTerm))
-      )
+      ? gameList.filter((game) => {
+          return (
+            (filterCriteria.searchTerm === NULL_SEARCH_TERM ||
+              cleanWord(game.defaultName).includes(
+                cleanWord(filterCriteria.searchTerm)
+              )) &&
+            hasIntervalOverlap(
+              [filterCriteria.nbOfPlayersMin, filterCriteria.nbOfPlayersMax],
+              game.playerCount
+            ) &&
+            hasIntervalOverlap(
+              [filterCriteria.durationMin, filterCriteria.durationMax],
+              game.duration
+            ) &&
+            (game.materialLanguage === MULTI_LANG_KEYWORD ||
+              (Object.keys(filterCriteria.language).includes(
+                game.materialLanguage
+              ) &&
+                filterCriteria.language[game.materialLanguage] === true))
+          );
+        })
       : [];
-  }, [gameList, searchTerm]);
+  }, [gameList, filterCriteria]);
 
   if (isLoading) {
     return null;
   }
 
+  const onChangeNbOfPlayersSlider = function (values) {
+    setFilterCriteria({
+      ...filterCriteria,
+      nbOfPlayersMin: values[0],
+      nbOfPlayersMax: values[1],
+    });
+  };
```

Comment:
With the state modification it becomes:

```suggestion
  const onChangeNbOfPlayersSlider = function (values) {
    setFilterCriteria({
      ...filterCriteria,
      nbOfPlayers: values,
    });
  };
```

Reply:
I prefer less "elegant" and more explicit.